test/message-generator/mock/template-provider-mock-quick-template.json

```diff
@@ -0,0 +1,215 @@
+                "n_bits": 545259519,
+                "target": [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,255,255,127]
+            },
+            "id": "set_new_prev_hash2"
+        },
+        {
+            "message": {
+                "type": "RequestTransactionDataSuccess",
+                "template_id": 30,
+                "excess_data": [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],
+                "transaction_list": []
+            },
+            "id": "request_transaction_data_success2"
+        }
+    ],
+    "frame_builders": [
+        {
+            "type": "automatic",
+            "message_id": "test/message-generator/messages/common_messages.json::setup_connection_success_template_distribution"
+        },
+        {
+            "type": "automatic",
+            "message_id": "new_template"
+        },
+        {
+            "type": "automatic",
+            "message_id": "set_new_prev_hash"
+        },
+        {
+            "type": "automatic",
+            "message_id": "new_template2"
+        },
+        {
+            "type": "automatic",
+            "message_id": "set_new_prev_hash2"
+        },
+        {
+            "type":"automatic",
+            "message_id": "request_transaction_data_success"
+        },
+        {
+            "type":"automatic",
+            "message_id": "request_transaction_data_success2"
+        }
+    ],
+    "actions": [
+        {
+            "message_ids": [],
+            "role": "server",
+            "results": [
+                {
+                    "type": "match_message_type",
+                    "value": "0x00"
+                }
+            ],
+            "actiondoc": "(pool) This action checks that a SetupConnection message is received"
+        },
+        {
+            "message_ids": ["setup_connection_success_template_distribution"],
+            "role": "server",
+            "results": [],
+            "actiondoc": "(pool) This action sends SetupConnection.Success with flag 2 (TD protocol)"
+        },
+        {
+            "message_ids": [],
+            "role": "server",
+            "results": [
+                {
+                    "type": "match_message_type",
+                    "value": "0x70"
+                }
+            ],
+          "actiondoc": "This action checks that a message SetCoinbaseDataSize is received"
+        },
+        {
+            "message_ids": ["new_template","set_new_prev_hash"],
+            "role": "server",
+            "results": [],
+            "actiondoc": "This action sends NewTemplate and SetNewPrevHash"
+        },
+        {
+            "message_ids": [],
+            "role": "server",
+            "results": [
+                {
+                    "type": "match_message_type",
+                    "value": "0x73"
+                }
+            ],
+            "actiondoc": "This action checks RequestTransactionData received"
```

Comment:
as mentioned above, the mock TP sends `NewTemplate` and `SetNewPrevHash` before the JDC has had the time to bootstrap and establish a connection

as a consequence, JDC is never aware of neither of these messages, and it never really sends a `RequestTransactionData`, so this action keeps hanging and the mock TP doesn't progress